typescript/widgets/src/walletIntegrations/multiProtocol.tsx

```diff
@@ -127,14 +128,55 @@ export function getAccountAddressForChain(
   if (!chainName || !accounts) return undefined;
   const protocol = multiProvider.getProtocol(chainName);
   const account = accounts[protocol];
-  if (protocol === ProtocolType.Cosmos) {
+  if (
+    protocol === ProtocolType.Cosmos ||
+    protocol === ProtocolType.CosmosNative
+  ) {
     return account?.addresses.find((a) => a.chainName === chainName)?.address;
   } else {
     // Use first because only cosmos has the notion of per-chain addresses
     return account?.addresses[0]?.address;
   }
 }
 
+export function getAddressFromAccountAndChain(
+  account?: AccountInfo,
+  chainName?: ChainName,
+) {
+  if (!account || !chainName) {
+    return 'Unknown';
+  }
+
+  // by default display the first address of the account
+  let address = account.addresses[0]?.address ?? 'Unknown';
```

Comment:
Instead of doing `let` here, I'd rather do the if clause for cosmos first, and directly return in there, and then otherwise just return this, mutable state i always find much harder to reason about